src/components/form/FieldBase.vue

```diff
@@ -0,0 +1,263 @@
+<template>
+  <div
+    class="field"
+    :class="{
+      ['field--errored']: hasError,
+      ['field--active']: active,
+      ['field--filled']: filled,
+      ['field--minimized-label']: minimizedLabel,
+      ['field--disabled']: disabled,
+      ['field--grouped']: grouped,
+    }"
+  >
+    <div
+      v-if="icon"
+      class="field__icon"
+    >
+      <HdIcon
+        :src="icon"
+        role="presentation"
+      />
+    </div>
+    <div class="field__body">
+      <div class="field__main">
+        <slot v-bind="{ hasError }" />
+        <label
+          v-if="label"
+          :for="name"
+          class="field__label"
+        >
+          {{ label }}
+        </label>
+        <div class="field__input-right">
+          <slot name="input-right"/>
+        </div>
+        <div
+          v-if="!grouped"
+          class="field__border"
+        />
+      </div>
+      <p
+        class="field__helper"
+        v-html="helperText"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
+
+export default {
+  name: 'FieldBase',
+  components: {
+    HdIcon,
+  },
+  props: {
+    name: {
+      type: String,
+      required: true,
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    error: {
+      type: String,
+      default: '',
+    },
+    helper: {
+      type: String,
+      default: '',
+    },
+    active: {
+      type: Boolean,
+      default: false,
+    },
+    filled: {
+      type: Boolean,
+      default: false,
+    },
+    minimizedLabel: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    grouped: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    helperText() {
+      // We default to an invisible character to keep the same element height
+      // and avoid content shift
+      return this.error || this.helper || '&#8205;';
+    },
+    hasError() {
+      return Boolean(this.error);
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
```

Comment:
Shouldn't we avoid scoped css here?
As mentioned on issue #377 , we should try to avoid scoped css to make overriding styles easier.
Specially in this case, where we have a base for the input fields.

Reply:
I think this is a bigger topic and we have to tackle it at once (and not scope some styles and some not).
The main reason is that we have to think of a more namespaced classnames, such as `hd-field` instead of `field` which is more likely to be used in other projects and might result in styles leakage.

So I'd say we keep it out of this PRs scope.